Dapper.Contrib/SqlMapperExtensions.cs

```diff
@@ -657,19 +702,58 @@ public int Insert(IDbConnection connection, IDbTransaction transaction, int? com
     {
         var cmd = String.Format("insert into {0} ({1}) values ({2})", tableName, columnList, parameterList);
         connection.Execute(cmd, entityToInsert, transaction, commandTimeout);
-        var r = connection.Query("select @@IDENTITY id", transaction: transaction, commandTimeout: commandTimeout);
+        var r = connection.Query("select @@IDENTITY id", transaction: transaction, commandTimeout: commandTimeout).ToList();
+
+        if (r.First().id == null) return 0;
+        var id = (int) r.First().id;

```

Comment:
I'm not following how this would work, e.g. if the ID column is a GUID a lot of these chains still assume integer behavior. We'd only really be supporting `int` and "less than `int`"...or am I missing something?